File: algorithms/javaalgo/src/test/java/leetcode/number101to200/number101to110/number110/BalancedBinaryTreeTest.java
package leetcode.number101to200.number101to110.number110;

import leetcode.util.TreeNode;

public class BalancedBinaryTreeTest {

    private boolean answer = true;

    public boolean isBalanced(TreeNode root) {
        maxDepth(root);
        return answer;
    }

    private int maxDepth(TreeNode root) {
        if (!answer) {
            return 0;
        }
        if (root == null) {
            return 0;
        }
        int left = maxDepth(root.left);
        int right = maxDepth(root.right);
        if (Math.abs(left - right) > 1) {
            answer = false;
        }

        return Math.max(left, right) + 1;
    }
}

File: algorithms/javascriptalgo/leetcode/number101-200/number101-110/number110/balanced-binary-tree.test.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
const isBalanced = function(root) {
  let answer = true;

  const maxDepth = (root) => {
    if (!answer) return 0;
    if (!root) return 0;

    let left = maxDepth(root.left);
    let right = maxDepth(root.right);
    if (Math.abs(left - right) > 1) {
      answer = false;
    }
    return Math.max(left, right) + 1;
  }

  maxDepth(root);
  return answer;
};
